### DISCUSSION TITLE
Simple Polymorphic Example

### DISCUSSION THREAD

**Discussion by theovier** [pdvrieze/xmlutil#70] [Q&A] [answered]
Hi,

I need some help with a basic polymorphic example. 

Suppose we have 2 classes, ```Bar``` and ```Buzz``` which both implement the ```Foo``` interface.  Code-wise this could look like this:

```Kotlin

@Serializable
class Parent(val components: List<@Polymorphic Foo> = listOf())

interface Foo

@Serializable
class Bar(val name: String) : Foo

@Serializable
class Buzz(val id: Int) : Foo
```

Now let's assume the following XML:

```XML
<Parent>
    <Bar name="alice" />
    <Buzz id="1" />
</Parent>
```

With the above code I am unable to parse this XML file as I am hit with the ```UnknownXmlFieldException: Could not find a field for name Bar```. What am I doing wrong here?

Thanks

**Comment by theovier** [comment, marked as answer]
After some digging I found this [test case](https://github.com/pdvrieze/xmlutil/blob/46fd66dbff28e54505341fb8e4a7bebbd52bf259/serialization/src/commonTest/kotlin/nl/adaptivity/xml/serialization/AClassWithMultipleChildren.kt) which shows exactly how to achieve this basic polymorphism.

**Reply by pdvrieze** [reply]
Sounds good. You do need to specify a module (there is an exception to that with the `@XmlPolyChildren` annotation that does basically the same (it was created when serializermodules didn't exist).
